### DISCUSSION TITLE
is Refresh access tokens supported?

### DISCUSSION THREAD

**Discussion by edardev** [ory/kratos#2193] [Q&A] [answered]
For JWT authentication and authorization should I use Kratos or Hydra? and is it refresh token supported ?

**Comment by vinckr** [comment]
Ory Kratos implements [two types of flows](https://www.ory.sh/docs/kratos/self-service), [browser flows](https://www.ory.sh/docs/kratos/self-service#browser-flows-for-client-side-apps-single-page-apps-reactjs-angularjs-nextjs-) and [API flows](https://www.ory.sh/docs/kratos/self-service#api-flows). 
You are probably looking for the latter. 

Please also have a look at this document: [Do You Need OAuth2?](https://www.ory.sh/docs/hydra/concepts/before-oauth2)

**Comment by aeneasr** [comment, marked as answer]
Thank you @vinckr ! To extend your answer, there's helpful documentation on this now: https://www.ory.sh/docs/guides/session-management/overview#json-web-token-jwt-support

**Reply by vinckr** [reply]
Hey @edardev,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
